docs/source/llms/transformers/tutorials/prompt-templating/prompt-templating.ipynb

```diff
@@ -0,0 +1,348 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2024/01/16 17:28:42 WARNING mlflow.transformers: params provided to the `predict` method will override the inference configuration saved with the model. If the params provided are not valid for the pipeline, MlflowException will be raised.\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "inputs: \n",
+       "  [string (required)]\n",
+       "outputs: \n",
+       "  [string (required)]\n",
+       "params: \n",
+       "  ['max_new_tokens': long (default: 15)]"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import mlflow\n",
+    "\n",
+    "sample_input = \"Tell me the largest bird\"\n",
+    "params = {\"max_new_tokens\": 15}\n",
+    "signature = mlflow.models.infer_signature(\n",
+    "    sample_input,\n",
+    "    mlflow.transformers.generate_signature_output(generator, sample_input, params=params),\n",
+    "    params=params,\n",
+    ")\n",
+    "\n",
+    "# visualize the signature\n",
+    "signature"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Starting a new experiment\n",
+    "We create a new [MLflow Experiment](https://mlflow.org/docs/latest/tracking.html#experiments) so that the run we're going to log our model to does not log to the default experiment and instead has its own contextually relevant entry.\n",
+    "\n",
+    "### Logging the model with the prompt template\n",
+    "Logging the model using MLflow saves the model and its essential metadata so it can be efficiently tracked and versioned. We'll use `mlflow.transformers.log_model()`, which is tailored to make this process as seamless as possible. To save the prompt template, all we have to do is pass it in using the `prompt_template` keyword argument.\n",
+    "\n",
+    "Two important thing to take note of:\n",
+    "\n",
+    "1. A prompt template must be a string with exactly one named placeholder `{prompt}`. MLflow will raise an error if a prompt template is provided that does not conform to this format.\n",
+    "\n",
+    "2. `text-generation` pipelines with a prompt template will have the [return_full_text pipeline argument](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/inference_client#huggingface_hub.inference._text_generation.TextGenerationParameters.return_full_text) set to `False` by default. This is to prevent the template from being shown to the users, which could potentially cause confusion as it was not part of their original input. To override this behaviour, either set `return_full_text` to `True` via `params`, or by including it in a `model_config` dict in `log_model()`."
```

Comment:
Great call-out here. Can we add a condensed version of this notification here: https://www.mlflow.org/docs/latest/llms/transformers/index.html#important-details-to-be-aware-of-with-the-transformers-flavor as an additional bullet point? I think it's pretty important to mention :)